rever/activities/bibtex.xsh

```diff
@@ -0,0 +1,75 @@
+"""Activity for bumping version."""
+import datetime
+import os
+
+from rever import vcsutils
+from rever.activity import Activity
+try:
+    import bibtexparser
+except ImportError:
+    pass
+
+# $ACTIVITIES = ['bibtex']
+
+def render_authors(authors):
+    """Parse a list of of tuples of authors into valid bibtex
+
+    Parameters
+    ----------
+    authors: list of str
+        The authors eg ['Your name in nicely formatted bibtex'].
+        Please see ``<http://nwalsh.com/tex/texhelp/bibtx-23.html>`` for
+        information about how to format your name for bibtex
+
+    Returns
+    -------
+    str:
+        Valid bibtex authors
+    """
+    if len(authors) == 1:
+        return ' '.join(authors[0])
+    else:
+        return ' and '.join(authors)
+
+
+class BibTex(Activity):
+    """Writes a BibTex reference for the version of software
+
+    Minimal ``rever.xsh``::
+        '''
+        $PROJECT_NAME = <my_project>  # The name of your project
+        $AUTHORS = ['Name1', 'Name2']  # The name of the authors
+        $URL = <URL to Project>  # A URL to the code
+        '''
+    """
+
+    def __init__(self, *, deps=frozenset(('version_bump', )),
+                 bibfile='bibtex.bib'):
+        super().__init__(name='bibtex', deps=deps, func=self._func,
+                         desc="Write BibTex file for version")
+        self.bibfile = bibfile
+
+    def _func(self):
```

Comment:
This should be `_func(self, project_name=None, authors=(), url=None)` so that you get the envvars `$BIBTEX_PROJECT_NAME`, `$BIBTEX_AUTHORS`, `$BIBTEX_URL`.

Reply:
filename should probably also be configurable here, rather than in the init